src/app/mauro/data-schema-service.ts

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022-2023 University of Oxford
+and Health and Social Care Information Centre, also known as NHS Digital
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import { Injectable } from '@angular/core';
+import { DataClass, DataElement, DataModel } from '@maurodatamapper/mdm-resources';
+import { forkJoin, map, Observable, switchMap } from 'rxjs';
+import {
+  DataClassWithElements,
+  DataElementSearchResult,
+  DataRequest,
+  DataSchema,
+} from '../data-explorer/data-explorer.types';
+import { DataModelService } from './data-model.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DataSchemaService {
+  constructor(private dataModels: DataModelService) {}
+
+  dataRequestClasses(dataSchemas: DataSchema[]): DataClassWithElements[] {
```

Comment:
I think the 3 functions `dataRequestClasses()`, `dataRequestElements()` and `dataSchemaElements()` could have better names, I'm not sure what they are doing on first glance.

Reply:
I renamed these to include the prefix "get" as suggested in a couple of comments above.